sbtStudentCommands/Pssr.scala.template

```diff
@@ -36,18 +37,20 @@ object Pssr {
     savedExercises
   }
 
-  def pullSolution = Command.command("pullSolution") { state =>
+  def pullSolution: Command = Command.command("pullSolution") { state =>
     val currentExercise = getCurrentExercise(state)
-    val currentExerciseMain = new sbt.File(getRootFolder(state), "exercises/src/main")
-    val cueFolder = getCueFolder(state)
-    val solFolder = new sbt.File(cueFolder, currentExercise + "/src/main")
-    IO.delete(currentExerciseMain)
-    IO.copyDirectory(solFolder, currentExerciseMain)
-    Console.println(Console.GREEN + "[INFO] " + Console.RESET + s"Solution for exercise $currentExercise pulled successfully")
-    state
+    withZipFile(state, currentExercise) { () =>
+      val currentExerciseMain = new sbt.File(getRootFolder(state), "exercises/src/main")
+      val cueFolder = getCueFolder(state)
+      val solFolder = new sbt.File(cueFolder, currentExercise + "/src/main")
+      IO.delete(currentExerciseMain)
+      IO.copyDirectory(solFolder, currentExerciseMain)
+      Console.println(Console.GREEN + "[INFO] " + Console.RESET + s"Solution for exercise $currentExercise pulled successfully")
```

Comment:
An interesting thought would be to refactor this so that you could write something like:

Console.println(green("[INFO]") + " Solution for ...")